### PR TITLE
Adding a first cut cheyenne-specific bash script for calling pytest. …

### PR DESCRIPTION
**Pull request by jmccreight**
…Go DI!

### CODE REVIEW

**Comment by jmccreight** [comment]
This script is super general purpose. 
There are a few rough edges
1) option to use standardized python env
2) gymnastics around the reference repo... not ideal currently (but convenience)
3) not currently wired the exe_cmd through for non-standard MPI (easy).

File: tests/local/cheyenne/model_test.sh [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+## Candidates branch to tag the test directory and optionally update the reference.
+branch_name="$(git symbolic-ref HEAD 2>/dev/null)" || branch_name="$(git rev-parse --short HEAD)"
+branch_name=${branch_name##refs/heads/}
+printf "\e[7;49;94mTesting branch: $branch_name in $candidate_dir\e[0m\n"
+output_dir=/glade/scratch/`whoami`/take_test_$(basename $domain_dir)_${branch_name}
+
+#-------------------------------------------------------
+# "Update" the reference dir? This makes the branch match that of
+# the candidate and it fetches the NCAR:branch_name
+# There are two potential pitfalls here,
```
Comment (kafitzgerald) [inline tests/local/cheyenne/model_test.sh]: I'm a little bit confused about why the default behavior here was chosen.  The first pitfall seems like it'll be pretty common and the automatic updating makes some assumptions for how repos are configured.

Comment (jmccreight) [reply tests/local/cheyenne/model_test.sh]: It's just what was convenient for me... it was good for most of my cases. Should be good for most CI-ish cases. But I'm open to better ideas

Comment (rcabell) [reply tests/local/cheyenne/model_test.sh]: I think ultimately this should be an overridable default. But ok as a starting point.

File: tests/local/cheyenne/model_test.sh [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+printf "\e[7;49;94mModule information\e[0m\n"
+module purge
+# Is this strict enough in the sense that things might be changing?
+module load  $compiler_module  $mpi  ncarcompilers  netcdf  ncarenv || exit 4
+module list
+
+#-------------------------------------------------------
+# Python Env
+# Need option to install standardized python env.
+# Look in to pipenv some day and a required.txt file or similar
```
[the remark this answers is not in the document]
Comment (rcabell) [reply tests/local/cheyenne/model_test.sh]: I think it's not a huge performance hit to create a venv and tear it down each time, with the benefit of reproducibility being worth the download/install time.

File: tests/local/cheyenne/model_test.sh (hunk 1 above)
Comment (jmccreight) [reply tests/local/cheyenne/model_test.sh]: Agreed.

File: tests/local/cheyenne/model_test.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmccreight) [reply tests/local/cheyenne/model_test.sh]: We are going to see if I can use katelyns env, then we can just point to that on cheyenne and maintain that. A bit speedier...

**Pull request merged**